python/ray/_private/gpu_object_manager.py

```diff
@@ -1,16 +1,59 @@
-from collections import namedtuple
-from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+from typing import TYPE_CHECKING, Any, Dict, List, NamedTuple, Optional, Tuple
 
+try:
+    import torch
+
+    _TORCH_AVAILABLE = True
+except ImportError:
+    _TORCH_AVAILABLE = False
+    torch = None
+
+import ray.util.collective as collective
+from ray._private.custom_types import TensorTransportEnum
 from ray._raylet import ObjectRef
 from ray.actor import ActorHandle
+from ray.experimental.collective import get_collective_groups
+from ray.util.collective.types import Backend
 
 if TYPE_CHECKING:
     import torch
 
-# GPUObjectMeta is a named tuple containing the source actor and tensor metadata.
-# The tensor metadata is a list of tuples, each containing the shape and dtype
-# of a tensor in the GPU object store.
-GPUObjectMeta = namedtuple("GPUObjectMeta", ["src_actor", "tensor_meta"])
+# Only define these mappings if torch is available
+if _TORCH_AVAILABLE:
+    TENSOR_TRANSPORT_TO_COLLECTIVE_BACKEND = {
+        TensorTransportEnum.NCCL: Backend.NCCL,
+        TensorTransportEnum.GLOO: Backend.TORCH_GLOO,
+    }
+
+    COLLECTIVE_BACKEND_TO_TORCH_DEVICE = {
+        Backend.NCCL: torch.device("cuda"),
+        Backend.TORCH_GLOO: torch.device("cpu"),
+    }
+else:
+    TENSOR_TRANSPORT_TO_COLLECTIVE_BACKEND = {}
+    COLLECTIVE_BACKEND_TO_TORCH_DEVICE = {}
+
+
+def _raise_torch_not_available_exception():
+    if not _TORCH_AVAILABLE:
+        raise ImportError(
+            "`tensor_transport` requires PyTorch. "
+            "Please install torch with 'pip install torch' to use this feature."
```

Comment:
In my memory, `pip install torch` will install CPU version PyTorch. Maybe we can just ask users to install torch without providing the instruction.